Fix flaky test in `tests/meltano/cli/test_state.py`

A flaky test was introduced by https://github.com/meltano/meltano/pull/9804:

```python
self = <test_state.TestCliState object at 0xff78971e81f0>, tmp_path = PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw1/test_set_from_string_file_mutu0')
cli_runner = <fixtures.cli.MeltanoCliRunner object at 0xff7895d6cfd0>

    def test_set_from_string_file_mutually_exclusive(
        self,
        tmp_path: Path,
        cli_runner: MeltanoCliRunner,
    ) -> None:
        """Test that mutually exclusive options are rejected."""
        filepath = tmp_path / "file.json"
        filepath.write_text("{}")
    
        result = cli_runner.invoke(
            cli,
            [
                "state",
                "set",
                "--force",
                "test-state-id",
                "--input-file",
                filepath.as_posix(),
                "{}",
            ],
        )
>       assert result.exit_code == 1
E       assert 2 == 1
E        +  where 2 = <Result SystemExit(2)>.exit_code

tests/meltano/cli/test_state.py:203: AssertionError
```

(from [a CI run](https://github.com/meltano/meltano/actions/runs/21733750320/job/62694086387?pr=9819#step:12:2529)).

Source:

https://github.com/meltano/meltano/blob/1a4af7ac9fb6027f94a0256a2e84d8acd5db8677/tests/meltano/cli/test_state.py#L182-L202

--- a/tests/meltano/cli/test_state.py
+++ b/tests/meltano/cli/test_state.py
@@ -179,6 +179,7 @@ class TestCliState:
                     assert_cli_runner(result)
                     assert state_service.get_state(state_id) == state_payload
 
+    @pytest.mark.usefixtures("project")
     def test_set_from_string_file_mutually_exclusive(
         self,
         tmp_path: Path,

